src/Microsoft.Identity.Web.DownstreamApi/DownstreamApiExtensions.cs

```diff
@@ -52,14 +54,50 @@ public static class DownstreamApiExtensions
         {
             _ = Throws.IfNull(services);
 
-            services.Configure<DownstreamApiOptions>(serviceName, configureOptions);
-
-            // https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
-            services.AddScoped<IDownstreamApi, DownstreamApi>();
             services.Configure(serviceName, configureOptions);
+            RegisterDownstreamApi(services);
+
             return services;
         }
 
+        private static void RegisterDownstreamApi(IServiceCollection services)
```

Comment:
Is it not possible to unit test this? Seems like you just need to register a few mock ITokenAcquisition / IDownstream APIs with different lifetimes, get them from the service collection and assert on their lifetime.

Reply:
Done. Thanks